### PR TITLE
feat: mf-6093 migrate rss3 api

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/MF-6094
https://mask.atlassian.net/browse/MF-6333

### CODE REVIEW

**Review by guanbinrui — APPROVED** [review]

File: packages/plugins/RSS3/src/SiteAdaptor/FeedFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    networks: {
+        display: 'flex',
+        gap: theme.spacing(1, 1.5),
+        flexWrap: 'wrap',
+    },
+}))
+
+export const FeedFilters = memo(
+    forwardRef<HTMLDivElement, BoxProps>(function FeedFilters(props: BoxProps, ref) {
```
Comment (Jack-Works) [inline packages/plugins/RSS3/src/SiteAdaptor/FeedFilters.tsx:46]: forwardRef has been deprecated. see https://react.dev/blog/2024/04/25/react-19#ref-as-a-prop

Comment (swkatmask) [reply packages/plugins/RSS3/src/SiteAdaptor/FeedFilters.tsx:46]: But mui is not compatible with that yet.

Comment (swkatmask) [reply packages/plugins/RSS3/src/SiteAdaptor/FeedFilters.tsx:46]: image

Comment (Jack-Works) [reply packages/plugins/RSS3/src/SiteAdaptor/FeedFilters.tsx:46]: it should be like this:

```js
// old
export default forwardRef<HTMLDivElement, BoxProps>(function (props: BoxProps, ref) {})

// new
export default function({ ...props, ref }: BoxProps & { ref: ... }) {}
```

if this does not work, that will be strange

**Pull request merged**